File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.js
function solution(t, p) {
    const len = p.length
    let answer = 0
    for(let i=0; i<t.length; i++) {
        if(t.substr(i, len).length !== len) return answer
        if(Number(p) >= Number(t.substr(i, len))) answer++
    }
    return answer
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(string t, string p) {
    int answer = 0;
    
    int numberOfDigit = p.size();
    
    for(int i=0; i<=t.size()-numberOfDigit; i++) {
        string sliceNum = t.substr(i, numberOfDigit);
        if(sliceNum <= p) {
            answer++;
        }
    }
    
    return answer;
}